### PR TITLE
ta: collated ta compilation

### PR DESCRIPTION
**Pull request by liushiwei007**
Sort out the common code compiled by ta,
abstract RISCV and ARM compile the common part of ta.

Reviewed-by: chenchaokai <chenchaokai@eswincomputing.com>

### CODE REVIEW

File: ta/mk/build-user-ta.mk [hunk 1]
```diff
@@ -66,7 +66,7 @@ additional-compile-deps := $(ta_dev_kit-files-include)
 include mk/compile.mk
 # Install TA libraries before in-tree TAs can be linked
 additional-link-deps := $(ta_dev_kit-files-lib)
-include  ta/arch/$(ARCH)/link.mk
+include  ta/arch/link.mk
```
Comment (jforissier) [inline ta/mk/build-user-ta.mk:69]: This does not make much sense does it? If it's under `arch` then it applies to some specific architecture.
Same for `link_shlib.mk` and `user_ta_header.c`. I suggest moving them higher up into `ta/`.

Comment (liushiwei007) [reply ta/mk/build-user-ta.mk]: > This does not make much sense does it? If it's under `arch` then it applies to some specific architecture. Same for `link_shlib.mk` and `user_ta_header.c`. I suggest moving them higher up into `ta/`.

done

**Review by jforissier — APPROVED** [review]
`Reviewed-by: Jerome Forissier <jerome.forissier@linaro.org>`

Please replace `ta` with `TA` in the subject and description except for the `ta:` subsystem prefix which is fine in lowercase

**Comment by jforissier** [comment]
And perhaps `build: ta: add RISC-V linker script` would be more descriptive?

**Pull request merged**